ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java

```diff
@@ -1040,10 +1042,13 @@ public void testDoesNotGatherStatsIfCompactionFails() throws Exception {
 
     txnHandler.compact(new CompactionRequest("default", "mtwb", CompactionType.MINOR));
 
+    CompactorFactory mockedFactory = Mockito.mock(CompactorFactory.class);
+    when(mockedFactory.getQueryCompactor(any(), any(), any(), any())).thenThrow(new RuntimeException());
+
     Worker worker = Mockito.spy(new Worker());
-    Mockito.when(worker.getMrCompactor()).thenThrow(RuntimeException.class);
     worker.setConf(conf);
     worker.init(new AtomicBoolean(true));
+    Whitebox.setInternalState(worker, "compactorFactory", mockedFactory);
```

Comment:
How about adding a constructor which takes the `compactorFactory` as a parameter?